### DISCUSSION TITLE
Turtlebot diagnostic error: IMU frequency is too high

### DISCUSSION THREAD

**Discussion by ValkyrX** [iRobotEducation/create3_docs#430] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Ubuntu 22.04

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H 02

### Which RMW is your robot running?

FastRTPS

### Is the robot connected to a network? If so, what is the network type?

Yes wifi network

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

Yes

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

Hi, when running diagnostics on the robot i get a warning that the IMU is too high ? is this a known issue, will it cause issues ? and is there a way to calibrate the IMU etc ?

Thanks

**Comment by shamlian** [comment]
It would be helpful if you could please provide the raw logs (or at least, the exact text to which you are referring) so that we can examine the exact error message, but I am not aware of any issues involving the robot's IMU. Could you also please confirm the robot's firmware version?

**Comment by ValkyrX** [comment]
Thanks for reply, so when i run the diagnostics with ros2 turtlebot .py file i get the following ..

Full Name: /Turtlebot4/Create3/IMU/turtlebot4_diagnostics:  imu topic status
Component: turtlebot4_diagnostics:  imu topic status
Hardware ID: Turtlebot4
Level: WARNING
Message: Frequency too high.

Events in window: 998
Events since startup: 34517
Duration of window (s): 10.000819
Actual frequency (Hz): 99.791831
Minimum acceptable frequency (Hz): 54.900000
Maximum acceptable frequency (Hz): 69.300000

**Reply by shamlian** [reply, marked as answer]
This is a problem with TurtleBot 4's diagnostic script; there is nothing wrong with the Create 3 robot. I suspect you are running an old diagnostic script (or the TB4 team didn't update theirs). Create 3's IMU update rate was increased to 100 Hz some time ago to improve accuracy. (This is a good thing.) Future questions about TurtleBot 4 in particular should likely go [here](https://github.com/turtlebot/turtlebot4/issues/).

**Comment by ValkyrX** [comment]
Thats great news , i did wonder where to post to be honest, just that as the create 3 holds the IMU i would try here 1st

Many Thanks

**Reply by shamlian** [reply]
No worries at all, happy to (try to) help! :-D
